Scripts/Chapter.js

```diff
@@ -151,5 +153,25 @@ function FK_ToggleMinWidth(disable) {
  * @return {boolean} True if image is doubled (width >= height) false otherwise
  */
 function FK_IsDoublePage(img) {
+	if (FK_IsRightWidth(img)){
+		return false;
+	}
 	return (img.naturalWidth >= img.naturalHeight);
-}
\ No newline at end of file
+}
+
+/**
+ * Check if the image has same width as the rest of the chapter
+ * @param {javascript node} img - The image to test
+ * @return {boolean} True if image is same size false otherwise
+ */
+function FK_IsRightWidth(img) {
+	/*if (typeof variable === 'undefined' || variable === null) {
+		var chapWidth = 0;
+	}*/
+	if (chapWidth == 0) {
+		chapWidth = img.naturalWidth;
+	} else if (chapWidth == img.naturalWidth){
+		return true;
+	}
+	return false;
+}
```

Comment:
I feel it would be cleaner to write

if (chapWidth == 0) {
    chapWidth = img.naturalWidth;
    return false;
}
return (chapWidth == img.naturalWidth);

This way the definition and the use of the variable is cleanly separated from one another.


On another note, I am not 100% certain as to what situation your code target exactly. Could you provide an explanation with a link to a chapter with an example?

Reply:
For instance here: https://kissmanga.com/Manga/Solo-Leveling/Chapter-096?id=571816
and some other such series.
The issue is that some series were made to be in a long strip, continuous scroll format. When the chapters are released, they aren't given as one single long image, but rather as pieces. These pieces are not of uniform size, and therefore some of them are wider than they are tall, making FreeKiss detect them as a double-page and resizing them to a different width from the rest of the pages, resulting in a broken appearance.

Reply:
Oh, ok, I didn't know they sliced webcomic this way. Is there an example in the link you provided or was it just to show a webcomic format? Because I didn't see any page out of proportion. Maybe I missed it.

Reply:
yes, there was an example there. Attaching some pics showing it from a couple different series (zoomed out):

![image](https://user-images.githubusercontent.com/4873274/70379322-2bdc4280-18f9-11ea-9433-0ea4bfc39b81.png)
![image](https://user-images.githubusercontent.com/4873274/70379325-35fe4100-18f9-11ea-8e35-b24347e34a29.png)
![image](https://user-images.githubusercontent.com/4873274/70379328-3eef1280-18f9-11ea-9d08-c40e2d0578c8.png)
![image](https://user-images.githubusercontent.com/4873274/70379330-431b3000-18f9-11ea-834b-3f63767f3733.png)
![image](https://user-images.githubusercontent.com/4873274/70379333-4d3d2e80-18f9-11ea-964b-55f20bce87e5.png)

There are some others.
When I did this I noticed that some of the previous ones were no longer having the issue, but new panels from the same chapters were - so maybe kiss has different servers with different images? I'm not sure why it's not being consistent, but from several days use the fix seems to function without issue.

Reply:
Ok, I didn't see the example on the link you provided, but I know KissManga is a big mess, so it is quite possible they have multiple servers which serve images cut differently.
We're good to go then ! Thanks ! I'll update FreeKiss next week with the changes :)

Reply:
Thanks!